backstory/src/main/java/com/google/sps/perspective/PerspectiveManager.java

```diff
@@ -73,10 +91,27 @@ public PerspectiveManager(PerspectiveAPI perspectiveAPI) {
    * @return An object describing the recommendation resulting from the analysis.
    * @throws NoAppropriateStoryException if story is not considered appropriate
    */
-  public StoryDecision generateDecision(String story) {
-    // TODO: Replace this fake behavior with a real API call, etc.
+  public StoryDecision generateDecision(String story) throws NoAppropriateStoryException {
+    PerspectiveAPIClient apiClient = new PerspectiveAPIClient(perspectiveAPI);
+    PerspectiveValues storyValues = apiClient.analyze(REQUESTED_ATTRIBUTES, story);
+    boolean decision = ContentDecisions.makeDecision(storyValues);
 
-    // returns default instance of StoryDecision
-    return new StoryDecision(story);
+    // if content decisions returns that it's appropriate (true),
+    // then return a StoryDecision object with this story
+    if (decision) {
+      return new StoryDecision(story);
+    }
+
+    // otherwise throw the NoAppropriateStoryException;
+    throw new NoAppropriateStoryException("The story passed in was not appropriate.");
+  }
+
+  /**
+   * Gets an array of the AttributeTypes that will be analyzed (mainly for testing).
+   *
+   * @return An array of the AttributeTypes we're requesting Perspective to analyze
+   */
+  public static AttributeType[] getRequestedAttributes() {
+    return REQUESTED_ATTRIBUTES;
```

Comment:
Since this is a constant, is there any advantage of creating a separate getter? Seems like doing PerspectiveManager.REQUESTED_ATTRIBUTES would serve the same purpose but avoid extra computation.

Reply:
This is for the purposes of testing (I wanted to check that we're getting the right number of attributes). I can also just make this public. I'm happy to do either. @ykjchen Which would you prefer?

Reply:
Done